server/consumer.go

```diff
@@ -330,7 +330,24 @@ func (mset *stream) addConsumerWithAssignment(config *ConsumerConfig, oname stri
 
 	// Make sure any partition subject is also a literal.
 	if config.FilterSubject != _EMPTY_ {
-		if !mset.validSubject(config.FilterSubject) {
+		subjects, cfg := mset.subjects(), mset.config()
+		var hasExt bool
+		if cfg.Mirror != nil {
+			if cfg.Mirror.External != nil {
+				hasExt = true
+			} else {
+				subjects = acc.streamSourceSubjects(cfg.Mirror)
+			}
+		} else if len(cfg.Sources) > 0 {
+			for _, si := range cfg.Sources {
+				if si.External != nil {
+					hasExt = true
+					break
+				}
+				subjects = append(subjects, acc.streamSourceSubjects(si)...)
```

Comment:
In case we have stream that it itself is sourced from another stream like:

```
Stream M1 <-OriginA
Stream M2 <- OriginB
Stream S1 <- M1, M2
```

Then it will fail to match since we need to check against the subjects from either OriginA or OriginB.

Reply:
Good point, need to make it recursive it seems to some degree.